### PR TITLE
Special char escaping

### PR DESCRIPTION
**Pull request by jlgeering**
---

#### Requirements

- [x] Entry in CHANGELOG.md was created
- [ ] Link to documentation on https://yaml.org/ is provided in the PR description
- [x] Functionality is covered by newly created tests

### CODE REVIEW

**Comment by mruoss** [comment]
reminder (from the docs):

> Note that escape sequences are only interpreted in [double-quoted scalars](https://yaml.org/spec/1.2.2/#double-quoted-style). In all other [scalar styles](https://yaml.org/spec/1.2.2/#node-styles), the “\” character has no special meaning and non-[printable](https://yaml.org/spec/1.2.2/#character-set) characters are not available.

File: test/ymlr/encode_test.exs [hunk 1]
```diff
@@ -79,8 +80,8 @@ defmodule Ymlr.EncodeTest do
     end
 
     test "quoted strings - escape seq forces double quotes (tab char)" do
-      assert_identity_and_output("a\tb", ~s("a\tb"))
-      assert_identity_and_output("!a\tb", ~s("!a\tb"))
+      assert_identity_and_output("a\tb", ~s(a\tb))
+      assert_identity_and_output("!a\tb", ~s('!a\tb'))
```
Comment (jlgeering) [inline test/ymlr/encode_test.exs:84]: @mruoss  the test says "forces double quotes" - but you removed the double quotes ... 

=> still wip?

Comment (mruoss) [reply test/ymlr/encode_test.exs:84]: Yeah.... did not really look at the test name. => should be renamed.

Comment (mruoss) [reply test/ymlr/encode_test.exs:85]: there. fixed.

File: BENCHMARK.md [hunk 2]
```diff
@@ -65,20 +65,20 @@ Run Time
...
 
   <tr>
     <td style="white-space: nowrap">Ymlr</td>
-    <td style="white-space: nowrap; text-align: right">22.80</td>
-    <td style="white-space: nowrap; text-align: right">43.87 ms</td>
-    <td style="white-space: nowrap; text-align: right">&plusmn;2.48%</td>
-    <td style="white-space: nowrap; text-align: right">43.77 ms</td>
-    <td style="white-space: nowrap; text-align: right">48.11 ms</td>
+    <td style="white-space: nowrap; text-align: right">1.58</td>
+    <td style="white-space: nowrap; text-align: right">632.01 ms</td>
```
Comment (mruoss) [inline BENCHMARK.md:78]: Yeah ok. I'll address this problem in a separate PR.

**Pull request merged**